__tests__/app.test.js

```diff
@@ -176,4 +174,88 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+  test("should respond with 201", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "testing comment",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201);
+  });
+  test("should respond the posted conmment", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "testing comment",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
+          comment_id: expect.any(Number),
+          body: "testing comment",
+          article_id: expect.any(Number),
+          author: "butter_bridge",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
 
+  test("status:400, responds with an error message when passed no user ID or no comment body", () => {
+    const newComment = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid input");
+      });
+  });
+  test("status:404, responds with an error message when passed a unknown user ID", () => {
+    const newComment = {
+      username: "Karo",
+      body: "testing comment",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found!");
+      });
+  });
+  test("status:404, responds with an error message when passed a non existent article id", () => {
+    const newComment = {
+      username: "Karo",
```

Comment:
You'd have to update the username to be one that exists in order to be sure it's the article and not the user that is not found